bindings/python/setup.py

```diff
@@ -153,32 +167,85 @@ def write_b2_python_config(
 
 class LibtorrentBuildExt(build_ext_lib.build_ext):
 
+    CONFIG_MODE_DISTUTILS = "distutils"
+    CONFIG_MODE_B2 = "b2"
+    CONFIG_MODES = (CONFIG_MODE_DISTUTILS, CONFIG_MODE_B2)
+
     user_options = build_ext_lib.build_ext.user_options + [
+        (
+            "config-mode=",
+            None,
+            "'b2' or 'distutils' (default). "
+            "In b2 mode, setup.py will just invoke b2 using --b2-args. "
+            "It will not attempt to auto-configure b2 or override any "
+            "args. "
+            "In distutils mode, setup.py will attempt to configure "
+            "and invoke b2 to match the expectations and behavior of "
+            "distutils (libtorrent will be built against the invoking "
+            "python, etc; note not all behaviors are currently supported). "
+            "The feature set will match the version found on pypi. "
+            "You can selectively override the auto-configuration in "
+            "this mode with --no-autoconf or --b2-args. For example "
+            "--b2-args=python=x.y or --no-autoconf=python will prevent "
+            "python from being auto-configured",
+        ),
+        (
+            "b2-args=",
```

Comment:
shouldn't `b2-args` be singular? you can only provide one argument to b2 at a time, right? i.e. `b2-arg`

Reply:
I couldn't find a reasonable way to pass multiple args here. The only way to override the option parsing mechanism requires overriding some large functions from `distutils` in a subclass, and changing one line out of a few dozen.

So `--b2-args` takes the whole argument sequence as a single string. We parse it with `shlex`, which allows spaces within args to b2, like: `--b2-args 'my-feature="a value with spaces"'`.

I'll update its documentation string to make that a little more clear.